build.gradle

```diff
@@ -84,6 +86,72 @@ check.doLast {
 }
 /////////////////// END jmh /////////////////////
 
+/////////////////////////////////////////////////////////////////
+//                   SonaType Central Repo                     //
+/////////////////////////////////////////////////////////////////
+/*
+task javadocJar(type: Jar) {
+    classifier = 'javadoc'
+    from javadoc
+}
+*/
+
+task sourcesJar(type: Jar) {
+    classifier = 'sources'
+    from sourceSets.main.allSource
+}
+
+artifacts {
+    archives sourcesJar
+}
+
+signing {
+    sign configurations.archives
```

Comment:
If you add:

        required { gradle.taskGraph.hasTask("uploadArchives") }

to this closure, people will be able to run `gradle build` without an error. Otherwise, they will get:

> Cannot perform signing task ':signArchives' because it has no configured signatory

But as long as they are not uploading, they shouldn't be signing either.